src/common/src/util/chunk_coalesce.rs

```diff
@@ -202,20 +212,16 @@ impl DataChunkBuilder {
         }
     }
 
-    fn build_data_chunk(&mut self) -> DataChunk {
-        let mut new_array_builders = vec![];
-        swap(&mut new_array_builders, &mut self.array_builders);
+    pub fn build_data_chunk(&mut self) -> DataChunk {
```

Comment:
To avoid accidentally yielding an empty chunk, we make this private, and callers should use `consume_all`. Does that meet your needs?

Reply:
Yes I think `consume_all` maybe better, let me fix it.